File: packages/typescript/src/resources/automation.ts
import type { HttpClient } from '../http/client.js';
import type { WorkflowList, WorkflowRun } from '../types/index.js';

export class WorkflowsResource {
    private readonly httpClient: HttpClient;
    private readonly basePath = '/v1/automation/workflows';

    constructor(httpClient: HttpClient) {
        this.httpClient = httpClient;
    }

    /**
     * List all workflows
     */
    async list(): Promise<WorkflowList> {
        const response = await this.httpClient.request<WorkflowList>({
            method: 'GET',
            path: this.basePath,
        });
        return response.data;
    }

    /**
     * Trigger a workflow
     */
    async trigger(id: string, params: Record<string, unknown> = {}): Promise<WorkflowRun> {
        const response = await this.httpClient.request<WorkflowRun>({
            method: 'POST',
            path: `${this.basePath}/${id}/trigger`,
            body: params,
        });
        return response.data;
    }
}

export class AutomationResource {
    public readonly workflows: WorkflowsResource;

    constructor(httpClient: HttpClient) {
        this.workflows = new WorkflowsResource(httpClient);
    }
}

File: packages/go/luna/resources/automation.go
package resources

import (
	"context"
	"encoding/json"
	"fmt"

	lunahttp "github.com/eclipse-softworks/luna-sdk-go/luna/http"
)

// WorkflowsResource provides access to workflow operations
type WorkflowsResource struct {
	client   *lunahttp.Client
	basePath string
}

// List retrieves all workflows
func (r *WorkflowsResource) List(ctx context.Context) (*WorkflowList, error) {
	resp, err := r.client.Request(ctx, lunahttp.RequestConfig{
		Method: "GET",
		Path:   r.basePath,
	})
	if err != nil {
		return nil, err
	}

	var result WorkflowList
	if err := json.Unmarshal(resp.Data, &result); err != nil {
		return nil, fmt.Errorf("failed to unmarshal response: %w", err)
	}

	return &result, nil
}

// Trigger triggers a workflow execution
func (r *WorkflowsResource) Trigger(ctx context.Context, id string, params any) (*WorkflowRun, error) {
	resp, err := r.client.Request(ctx, lunahttp.RequestConfig{
		Method: "POST",
		Path:   fmt.Sprintf("%s/%s/trigger", r.basePath, id),
		Body:   params,
	})
	if err != nil {
		return nil, err
	}

	var result WorkflowRun
	if err := json.Unmarshal(resp.Data, &result); err != nil {
		return nil, fmt.Errorf("failed to unmarshal response: %w", err)
	}

	return &result, nil
}

// AutomationResource groups Automation service resources
type AutomationResource struct {
	Workflows *WorkflowsResource
}

// NewAutomationResource creates a new Automation resource
func NewAutomationResource(client *lunahttp.Client) *AutomationResource {
	return &AutomationResource{
		Workflows: &WorkflowsResource{
			client:   client,
			basePath: "/v1/automation/workflows",
		},
	}
}
